src/create-cache.js

```diff
@@ -96,21 +220,28 @@ const addRefresh = (cache) => {
 }
 
 const overrideKeys = (cache, prefix) => {
-  const _keys = cache.store.keys;
-  cache.store.keys = async (pattern = "*", cb) => {
-    if (typeof pattern === 'function') {
-      cb = pattern;
-      pattern = '*';
-    }
-    logger.trace(`keys: invoked. prefix=${prefix}, pattern=${pattern}`);
-    pattern = prefix + pattern;
-    let keys = await _keys.apply(cache.store, [pattern]);
-    keys = keys.map((key) => key.substr(prefix.length));
-    if (cb) {
-      cb(keys);
-    }
-    return keys;
-  }
+  const redisClient = cache.store.getClient();
+
+  cache.store.keys = (pattern = "*", cb) => {
+    return new Promise((resolve, reject) => {
+      if (typeof pattern === 'function') {
+        cb = pattern;
+        pattern = '*';
+      }
+
+      logger.trace(`keys: invoked. prefix=${prefix}, pattern=${pattern}`);
+
```

Comment:
reject in advance if `pattern !== '*'`, as we don't support any other pattern for now.